view/frontend/web/js/action/express-pay/update-quote-shipping-method-action.js

```diff
@@ -20,23 +20,26 @@ define(
          * @return void
          */
         return async function (shippingMethod = null) {
-            let newMethod = null;
+            let newMethod;
+            let ms = 100;
+            let carry = 100;
+            while (shippingService.isLoading() && ms < 5000) {
+                // max total timeout is 8000ms
+                await new Promise(resolve => setTimeout(resolve, ms));
+                carry = ms + carry;
+                ms = carry - ms;
+            }
             shippingMethod = shippingMethod ?? quote.shippingMethod();
             if (shippingMethod !== null) {
-                let availableMethods = shippingService.getShippingRates().filter((method) => {
+                newMethod = shippingService.getShippingRates().filter((method) => {
                     let methodId = `${method.carrier_code}_${method.method_code}`;
                     methodId = methodId.replace(/\s/g, '');
-                    return methodId === shippingMethod['id'] || methodId === shippingMethod['identifier'];
-                });
-                if (availableMethods.length > 0) {
-                    newMethod = availableMethods[0];
-                }
-            } else {
-                newMethod = shippingService.getShippingRates().first();
-            }
-            if (newMethod !== null) {
-                selectShippingMethodAction(newMethod);
+                    return methodId === shippingMethod['id'] 
+                        || methodId === shippingMethod['identifier'] 
+                        || methodId === `${shippingMethod.carrier_code}_${shippingMethod.method_code}`;
+                }).first();
```

Comment:
so what if the shipping rate **is** NULL? I don't see we are handling this. What if the shipping rates are not properly configured in shop, will it crash GPay?

Reply:
I didn't do a test for no rates returned from the shipping service but if there are no matches in the filter, the `first()` function will return `undefined`. Then when the code gets to line 42 calling `selectShippingMethodAction` will be called with the first available shipping rate.

Reply:
ok, sounds good! thanks for confirming.

Reply:
Just checked for no available shipping rates for an address and the update callback fails and you have to select a different address, the modal does not crash.
You get this error message `The shipping method is missing. Select the shipping method and try again.`